src/main/java/command/AddDeadlineTaskCommand.java

```diff
@@ -0,0 +1,58 @@
+package com.leeyiyuan.command;
+
+
+import com.leeyiyuan.storage.Storage;
+import com.leeyiyuan.storage.StorageException;
+import com.leeyiyuan.task.DeadlineTask;
+import com.leeyiyuan.task.TaskList;
+import com.leeyiyuan.ui.UserOutputInterface;
+import java.time.LocalDateTime;
+
+/** 
+ * Represents a Command to add a DeadlineTask. 
+ */
+public class AddDeadlineTaskCommand extends Command {
+
+    /** 
+     * Title for the new DeadlineTask. 
+     */
+    protected String title;
+
+    /** 
+     * By value for the new DeadlineTask. 
+     */
+    protected LocalDateTime by;
+
+    /**
+     * Constructs an AddDeadlineTaskCommand from a title and a by value.
+     *
+     * @param title The title for the new DeadlineTask.
+     * @param by The by value for the new DeadlineTask.
+     */
+    public AddDeadlineTaskCommand(String title, LocalDateTime by) {
+        this.title = title;
+        this.by = by;
+    }
+
+    /** 
+     * {@inheritDoc} 
+     */
+    @Override
+    public void execute(TaskList tasks, UserOutputInterface uoi, Storage storage)
+            throws CommandExecuteException, StorageException {
+        if (tasks.containsTitle(this.title)) {
+            throw new CommandExecuteException("A task with the same title already exists.");
+        }
+
+        DeadlineTask task = new DeadlineTask();
```

Comment:
Have no issues, but would like to ask why you wanted to have setTitle and setBy methods instead of passing the title and 'by time' in the constructor

Reply:
Hi Gabriel, thanks for the kind comments. I had considered the possibility of future requirements that involve mutating the task objects, and have decided on the said design of making tasks mutable through the setter methods. Hope this clarifies!